chat/rich-links/src/RichLink/RichLink.types.ts

```diff
@@ -0,0 +1,55 @@
+import { DarkModeProps, HTMLElementProps } from '@leafygreen-ui/lib';
+
+import { Variant as RichLinkBadgeVariantName } from './RichLinkBadge/RichLinkBadge.types';
+import { RichLinkVariantName } from './RichLinkVariants';
+
+export interface BaseRichLinkProps
+  extends HTMLElementProps<'a', HTMLAnchorElement>,
+    DarkModeProps {
+  /**
+   * The URL that the rich link points to
+   */
+  url: string;
+
+  /**
+   * The text that shows on the rich link
+   */
+  text: string;
+
+  /**
+   * A URL for the background image of the rich link
+   */
+  imageUrl?: string;
+
+  /**
+   * Props to be spread on the anchor element
+   */
+  anchorProps?: Pick<JSX.IntrinsicElements['a'], 'target' | 'rel'>;
```

Comment:
I don't fully understand this prop. On L7, you're extending anchor props, which means you can pass `target` or `rel` outside of this prop. Can you explain the reason for this?

Reply:
I think you're right - a relic of working around some TS error I think. Removed this now I'm just picking off these named props and spreading rest on to `...anchorProps`.

```
const {
  url,
  text,
  imageUrl,
  badgeGlyph,
  badgeLabel,
  badgeVariant,
  ...anchorProps
} = props;
```

However that's revealing some weird TS errors. I think the gist of these is that `Card` is not properly typing based on the `as` prop. A bit stumped atm but I'll keep looking at it. Any ideas?

```
Type '{ children: (Element | null)[]; key?: Key | null | undefined; download?: any; href: string; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 270 more ...; as: "a"; }' is not assignable to type 'IntrinsicAttributes & ((Omit<BoxAnchorDefault<CardProps>, "ref"> | Omit<BoxIntrinsic<keyof IntrinsicElements, CardProps>, "ref"> | Omit<...> | Omit<...>) & RefAttributes<...>)'.
  Types of property 'as' are incompatible.
    Type '"a"' is not assignable to type 'undefined'.ts(2322)
```

```
 '{ children: (Element | null)[]; ref: ForwardedRef<HTMLAnchorElement>; key?: Key | null | undefined; download?: any; hrefLang?: string | undefined; ... 272 more ...; href: string; } | { ...; }' is not assignable to type 'IntrinsicAttributes & ((Omit<BoxAnchorDefault<CardProps>, "ref"> | Omit<BoxIntrinsic<keyof IntrinsicElements, CardProps>, "ref"> | Omit<...> | Omit<...>) & RefAttributes<...>)'.
  Type '{ children: (Element | null)[]; ref: ForwardedRef<HTMLAnchorElement>; key?: Key | null | undefined; download?: any; hrefLang?: string | undefined; ... 272 more ...; href: string; }' is not assignable to type 'IntrinsicAttributes & ((Omit<BoxAnchorDefault<CardProps>, "ref"> | Omit<BoxIntrinsic<keyof IntrinsicElements, CardProps>, "ref"> | Omit<...> | Omit<...>) & RefAttributes<...>)'.
    Property 'href' does not exist on type 'IntrinsicAttributes & Omit<BoxIntrinsic<keyof IntrinsicElements, CardProps>, "ref"> & RefAttributes<unknown>'. Did you mean 'ref'?ts(2322)
```

Reply:
Ended up removing the extends and going with the explicit `anchorProps` prop. Still need to use an `@ts-ignore` (☹️) because `Card` still thinks it's an `HTMLDivElement` instead of `HTMLAnchorElement`.

Reply:
https://jira.mongodb.org/browse/LG-4259

Reply:
Thanks for creating the ticket! 

If you don't add the `anchorProps` prop, would you still have to add `@ts-ignore`? Adding a prop just for anchor props isn't a pattern I'm used to seeing, and it sorta feels like an extra step for consumers. @bruugey Do you have any thoughts on this?

Reply:
Do you mean the explicit `BaseRichLinkProps.anchorProps` or passing additional anchor props to the link in general?

RE the explicit prop, I also played around with the rich link props extending `HTMLElementProps<'a', never>` but was still hitting TS errors. No matter how I get them from the user, there's still a mismatch between what Card and I expect the props to be.

In general, I think we'd want folks to be able to control link things like `target`, `rel`, etc if they need to

Reply:
Yeah, I was referring to the explicit `anchorProps` prop. Are you're saying that,
```
export interface BaseRichLinkProps extends DarkModeProps, Omit<HTMLElementProps<'a', never>, 'children' | 'href'> {}
```
causes more TS errors?

Reply:
@shaneeza Good call - updated!